src/foundry/client-esm/canvas/sources/point-vision-source.d.mts

```diff
@@ -0,0 +1,205 @@
+> extends PointEffectSourceMixin_RenderedEffectSource<SourceData> {
+  /** @defaultValue `"sight"` */
+  static override sourceType: string;
+
+  /** @defaultValue `["visionMode", "blinded"]` */
+  static override _initializeShaderKeys: string[];
+
+  /** @defaultValue `["radius", "color", "attenuation", "brightness", "contrast", "saturation", "visionMode"]` */
+  static override _refreshUniformsKeys: string[];
+
+  /**
+   * The corresponding lighting levels for dim light.
+   * @defaultValue `LIGHTING_LEVELS.DIM`
+   */
+  protected static _dimLightingLevel: foundry.CONST.LIGHTING_LEVELS;
+
+  /**
+   * The corresponding lighting levels for bright light.
+   * @defaultValue `LIGHTING_LEVELS.BRIGHT`
+   */
+  protected static _brightLightingLevel: foundry.CONST.LIGHTING_LEVELS;
+
+  /** @defaultValue `"visionSources"` */
+  static override effectsCollection: string;
+
+  /**
+   * @defaultValue
+   * ```js
+   * ...super.defaultData,
+   * contrast: 0,
+   * attenuation: 0.5,
+   * saturation: 0,
+   * brightness: 0,
+   * visionMode: "basic",
+   * lightRadius: null
+   * ```
+   */
+  static override defaultData: PointVisionSourceData;
+
+  static get _layers(): Record<string, RenderedEffectSource.RenderedEffectLayerConfig>;
+
+  /**
+   * The vision mode linked to this VisionSource
+   */
+  visionMode: VisionMode | null;
+
+  /**
+   * The vision mode activation flag for handlers
+   * @internal
+   */
+  _visionModeActivated: boolean;
+
+  /**
+   * The unconstrained LOS polygon.
+   */
+  los: PointSourcePolygon;
+
+  /**
+   * The polygon of light perception.
+   */
+  light: PointSourcePolygon;
+
+  /**
+   * An alias for the shape of the vision source.
+   */
+  get fov(): PIXI.Polygon | PointSourcePolygon;
+
+  /**
+   * If this vision source background is rendered into the lighting container.
+   */
+  get preferred(): boolean;
+
+  /**
+   * Is the rendered source animated?
+   */
+  get isAnimated(): boolean;
+
+  /**
+   * Light perception radius of this vision source, taking into account if the source is blinded.
+   */
+  get lightRadius(): number;
+
+  override get radius(): number;
+
+  /**
+   * Is this source temporarily blinded?
+   */
+  get isBlinded(): boolean;
+
+  /**
+   * Records of blinding strings with a boolean value.
+   * By default, if any of this record is true, the source is blinded.
+   */
+  blinded: Record<string, boolean>;
+
+  /**
+   * Data overrides that could happen with blindness vision mode.
+   */
+  visionModeOverrides: object;
+
+  override _initialize(data: Partial<SourceData>): void;
+
+  override _createShapes(): void;
+
+  override shape: PointSourcePolygon;
+
+  /**
+   * Responsible for assigning the Vision Mode and calling the activation and deactivation handlers.
+   */
+  protected _updateVisionMode(): void;
+
+  override _configure(changes: Partial<SourceData>): void;
+
+  override _configureLayer(layer: object, layerId: string): void;
```

Comment:
Probably only want plain objects here, i.e. `Record<string, unknown>`. `object` includes arrays and functions.

The fact that we use it elsewhere is either a bug (other people having this misconception) or in _very_ few places it's trying to avoid an index signature or sort around another specific type issue. I have this in the backlog of things for me to audit.